sequencer/src/block.rs

```diff
@@ -54,7 +54,8 @@ impl BlockPayload for Payload<TxTableEntryWord> {
         txs: impl IntoIterator<Item = Self::Transaction>,
         instance_state: &Self::Instance,
     ) -> Result<(Self, Self::Metadata), Self::Error> {
-        let payload = Payload::from_txs(txs, &instance_state.chain_config)?;
+        // TODO: ????
+        let payload = Payload::from_txs(txs, &instance_state.genesis_state.chain_config)?;
```

Comment:
@sveitser @jbearer This doesn't look right. Do we need to add ValidatedState type to this trait and pass that as the parameter to `from_transactions`

Reply:
Ugh yes we do (or equivalent). But the bigger problem is this means the builder will need to be aware of the ValidatedState at each point. This is fundamental because a parameter of block building (max size) is now being made variable.

In theory this is fine, the builder can compute the state by replaying each proposal it sees, but I don't know if it currently has all the information it needs from HotShot to call `validate_and_apply_header`. We'll need to discuss this with @move47 and @nyospe

Reply:
I don't believe we do, and I'm not sure it will be trivial to add.

Reply:
Even if we maintain the state as suggested, we will need a mechanism to recover when we miss a proposal...

Reply:
That part is built into the sequencer implementation of `ValidatedState`. `validate_and_apply_header` will automatically fetch missing state as needed as long as the `InstanceState` contains a valid peer URL